packages/manager/apps/pci-instances/src/pages/instances/instance/dashboard/hooks/__tests__/useDashboardAction.spec.tsx

```diff
@@ -0,0 +1,164 @@
+import { describe, it, vi } from 'vitest';
+import { renderHook, waitFor } from '@testing-library/react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { useUnattachedVolumes } from '../useDashboardAction';
+import { getInstance } from '@/data/api/instance';
+import { TInstance } from '@/types/instance/entity.type';
+import { getVolumes } from '@/data/api/volume';
+
+const fakeInstance = {
+  id: 'fake-instance-id',
+  name: 'fake-instance',
+  region: {
+    name: 'fake-region',
+    type: 'region',
+    availabilityZone: 'fake-zone-a',
+  },
+  volumes: [
+    {
+      id: 'fake-volume-1',
+      name: 'fake-volume',
+      size: null,
+    },
+  ],
+  task: {
+    isPending: false,
+    status: null,
+  },
+  actions: [{}],
+} as TInstance;
+
+const getVolumesMock = vi.fn();
+
+vi.mock('@/data/api/instance');
+vi.mock('@/data/api/volume');
+vi.mocked(getInstance).mockResolvedValue(fakeInstance);
+vi.mocked(getVolumes).mockImplementation(getVolumesMock);
+
+const render = () =>
+  renderHook(
+    () =>
+      useUnattachedVolumes({
+        projectId: 'fake-projectId',
+        region: 'fake-region',
+        instanceId: 'fake-instance',
+      }),
+    {
+      wrapper: ({ children }) => (
+        <QueryClientProvider client={new QueryClient()}>
+          {children}
+        </QueryClientProvider>
+      ),
+    },
+  );
+
+describe('useUnattachedVolumes', () => {
+  it.each([
```

Comment:
Question (non-blocking): why doing it like that ?

I find this way of testing to be harder to read and quite cluttered.